azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -349,44 +385,54 @@ def _get_telemetry_topic(self):
         topic += "/messages/events/"
         return topic
 
+    def _get_base_topic(self):
+        if self._auth_provider.module_id:
+            return (
+                "devices/"
+                + self._auth_provider.device_id
+                + "/modules/"
+                + self._auth_provider.module_id
+            )
+        else:
+            return "devices/" + self._auth_provider.device_id
+
     def _get_c2d_topic(self):
         """
         :return: The topic for cloud to device messages.It is of the format
         "devices/<deviceid>/messages/devicebound/#"
         """
-        return "devices/" + self._auth_provider.device_id + "/messages/devicebound/#"
+        return self._get_base_topic() + "/messages/devicebound/#"
 
     def _get_input_topic(self):
         """
         :return: The topic for input messages. It is of the format
         "devices/<deviceId>/modules/<moduleId>/messages/inputs/#"
         """
-        return (
-            "devices/"
-            + self._auth_provider.device_id
-            + "/modules/"
-            + self._auth_provider.module_id
-            + "/inputs/#"
-        )
+        return self._get_base_topic() + "/inputs/#"
 
     def connect(self, callback=None):
+        logger.info("connect called")
         self._connect_callback = callback
         self._trig_connect()
 
     def disconnect(self, callback=None):
+        logger.info("disconnect called")
         self._disconnect_callback = callback
         self._trig_disconnect()
 
     def send_event(self, message, callback=None):
-        self._trig_send_event(message, callback)
+        action = SendMessageAction(message, callback)
+        self._trig_queue_action(action, self._action_queue)
```

Comment:
>self._action_queue [](start = 40, length = 18)

here you have a second parameter but not below. leftovers? i don't see code that changes based on a second parameter?